packages/framework-integration-tests/integration/end-to-end/cart.integration.ts

```diff
@@ -162,6 +162,170 @@ describe('Cart end-to-end tests', () => {
     })
   })
 
+  describe('Entities', () => {
+    let userAuthToken: string
+    let userEmail: string
+    const mockPassword = 'Enable_G0d_Mode3e!'
+
+    before(async () => {
+      userEmail = internet.email()
+      // TODO: Make retrieval of auth token cloud agnostic
+      await createUser(userEmail, mockPassword, 'User')
+      userAuthToken = await getAuthToken(userEmail, mockPassword)
+      client = await graphQLClient(userAuthToken)
+    })
+
+    context('Reducers', () => {
+      let mockSku: string
+      let mockDisplayName: string
+      let mockDescription: string
+      let mockPriceInCents: number
+      let mockCurrency: string
+
+      let productId: string
+
+      beforeEach(async () => {
+        mockSku = random.uuid()
+        mockDisplayName = commerce.productName()
+        mockDescription = lorem.paragraph()
+        mockPriceInCents = random.number({ min: 1 })
+        mockCurrency = finance.currencyCode()
+
+        // Add one item
+        await client.mutate({
+          variables: {
+            sku: mockSku,
+            displayName: mockDisplayName,
+            description: mockDescription,
+            priceInCents: mockPriceInCents,
+            currency: mockCurrency,
+          },
+          mutation: gql`
+            mutation CreateProduct(
+              $sku: String!
+              $displayName: String
+              $description: String
+              $priceInCents: Float
+              $currency: String
+            ) {
+              CreateProduct(
+                input: {
+                  sku: $sku
+                  displayName: $displayName
+                  description: $description
+                  priceInCents: $priceInCents
+                  currency: $currency
+                }
+              )
+            }
+          `,
+        })
+
+        // Check that new product is available in read model
+        const products = await waitForIt(
+          () => {
+            return client.query({
+              query: gql`
+                query {
+                  ProductReadModels {
+                    id
+                    sku
+                    displayName
+                    description
+                    price
+                    availability
+                    deleted
+                  }
+                }
+              `,
+            })
+          },
+          (result) => result?.data?.ProductReadModels?.some((product: any) => product.sku === mockSku)
+        )
+
+        const product = products.data.ProductReadModels.find((product: any) => product.sku === mockSku)
+        productId = product.id
```

Comment:
I guess you use the ID returned in the "expected" result just because you are doing a deep comparison (as if not, it would fail), correct?

Reply:
that is correct @alvaroloes, I usually like checking it just in case 😅